tools/buildsys/src/args.rs

```diff
@@ -0,0 +1,214 @@
+/*!
+
+These structs provide the CLI interface for buildsys which is called from Cargo.toml and accepts all
+of its input arguments from environment variables.
+
+!*/
+
+use buildsys::manifest::SupportedArch;
+use clap::{Parser, Subcommand};
+use std::path::PathBuf;
+use url::Url;
+
+/// A list of environment variables and the type of build that should be rerun if that environment
+/// variable changes. The build type is represented with bit flags so that we can easily list
+/// multiple build types for a single variable. See `[BuildType]` and `[rerun_for_envs]` below to
+/// see how this list is used.
+const REBUILD_VARS: [(&str, u8); 13] = [
+    ("BUILDSYS_ARCH", PACKAGE | VARIANT),
+    ("BUILDSYS_NAME", VARIANT),
+    ("BUILDSYS_OUTPUT_DIR", VARIANT),
+    ("BUILDSYS_PACKAGES_DIR", PACKAGE),
```

Comment:
nit: it feels like a (pre-existing) mistake to not track `BUILDSYS_SOURCES_DIR` if we are tracking `BUILDSYS_PACKAGES_DIR` - if we override the build to point to a completely different sources tree then that should invalidate any builds that used those sources.

Reply:
> nit: it feels like a (pre-existing) mistake to not track `BUILDSYS_SOURCES_DIR` if we are tracking `BUILDSYS_PACKAGES_DIR` - if we override the build to point to a completely different sources tree then that should invalidate any builds that used those sources.

I'm going to defer this because I'm concerned that the symlink change when changing variant could trigger something. I'll track this as a GitHub issue and play around with adding it. https://github.com/bottlerocket-os/twoliter/issues/152